### PR TITLE
Add executor::block_on_sink

### PR DESCRIPTION
**Pull request by taiki-e**
Related: https://github.com/rust-lang-nursery/futures-rs/pull/1507#issuecomment-479585935

### CODE REVIEW

File: futures-executor/src/local_pool.rs [hunk 1]
```diff
@@ -334,6 +337,78 @@ impl<S: Stream + Unpin> Iterator for BlockingStream<S> {
...
+    /// the function returns `Pending` then the current thread is blocked
+    /// until it is otherwise ready to accept the value.
+    ///
+    /// # Return value
+    ///
+    /// If `Ok(())` is returned then the `value` provided was successfully sent
+    /// along the sink, and if `Err(e)` is returned then an error occurred
+    /// which prevented the value from being sent.
+    pub fn send(&mut self, item: Item) -> Result<(), S::Error> {
+        LocalPool::new().run_until(self.sink.send(item))
```
Comment (Matthias247) [inline futures-executor/src/local_pool.rs]: Why not just `block_on`? That should be cheaper.

Comment (Matthias247) [reply futures-executor/src/local_pool.rs]: But more of a general thing: I'm not sure if those wrappers are pullling their weight. Just using `block_on(sink.send(item))` is not much more boilerplate, works for all `Future`s, and doesn't require the original async type to get moved.

**Comment by cramertj** [comment]
Apologies for the delayed review-- I thought we already had this at one point, but perhaps it disappeared. I don't have any objections to adding it. @taiki-e do you want to update this PR and get it landed as part of 0.3?

File: futures-executor/src/local_pool.rs (hunk 1 above)
Comment (taiki-e) [reply futures-executor/src/local_pool.rs]: > Why not just `block_on`? That should be cheaper.

I just used the same thing as block_on_stream...
https://github.com/rust-lang-nursery/futures-rs/blob/25d638d720bfa128ff4f934d1742bfedfed20c89/futures-executor/src/local_pool.rs#L334

File: futures-executor/Cargo.toml [hunk 2]
```diff
@@ -18,9 +18,11 @@ name = "futures_executor"
 default = ["std"]
 std = ["futures-core-preview/std", "futures-util-preview/std"]
 threadpool = ["num_cpus"]
+sink = ["futures-sink-preview", "futures-util-preview/sink"]
```
Comment (taiki-e) [inline futures-executor/Cargo.toml]: Add a feature flag (same as -channel and -util).

**Comment by taiki-e** [comment]
I have no motivation to continue this. Just use `block_on(sink.send(item))`, as Matthias said.

**Pull request closed without merging**